### DISCUSSION TITLE
Connecting through instant messaging service

### DISCUSSION THREAD

**Discussion by vojtapolasek** [ComplianceAsCode/content#7064] [General]
Hello all,
currently, we are using #openscap IRC channel on Freenode for instant communication with other community members. As you probably know, there have been some possibly controversial things happening around Freenode recently. I think this can be a good time to reevaluate the choice of instant messaging platform for this project.
What would be your prefered way of instant communication within this project?
My 2 cents: I think it would be useful to use a platform which supports replies or even threading. Many contributors (and people who can merger PRs) are located in Europe. But there are also some contributors from other continents. There are long delays between questions and answers due to different time zones. Also some contributors are not always connected because not everyone has an IRC bouncer...
I was thinking about using Slack / Matrix. I know Slack can do threading, I am not sure about Matrix, but at least it can do replies.
What do others think?

**Comment by jan-cerny** [comment]
FYI there is #openscap on libera.chat.

**Reply by GaryGapinski** [reply]
I am open to the methods that have been mentioned. There is also gitter.im which supports threading.

**Comment by cipherboy** [comment]
I'd generally prefer IRC as I already have accounts and it is easier to join without a permanent account than Slack and Matrix are. With Slack in particular, free community plans end up hitting the message history limit, making them less useful than an IRC client to long-standing community members who like to search history. This might mean a paid Slack instance, I don't know.

I'm not likely to get another new account just for every community that wants to branch off from IRC -- I sit in approximately two dozen channels on Freenode. If every project I followed moved to a different Slack/... instance, I wouldn't re-join them all.

My 2c :-)

**Reply by jan-cerny** [reply]
I would also prefer using IRC because it has low entry barrier, most people in open source communities are already present on freenode or libera.chat and they can simply join the channel and they don't have to register another account at a new service.

If we continue using IRC for instant messaging I have 2 questions:
* Freenode or liberachat?
* Use `#openscap` or create a new channel for ComplianceAsCode?

**Reply by richardmaciel-canonical** [reply]
I also vote for IRC, but prefer a new channel just for ComplianceAsCode stuff. Don't have a preference on the IRC server.

**Comment by truzzon** [comment]
I'm still new to this whole community contribution thing. I have only tried to contribute to two projects so far. This one and salt.

So far I preferred Slack to communicate, because of the threading feature. The salt guys are mostly located in the US. So I can post my question, get on with my day and get a notification on my phone, when I received an answer. Very handy.

**Comment by matejak** [comment]
So plenty of projects seem to be using Gitter: https://gitter.im/home/explore
It is an IM, "without any limits", they claim interoperability with Matrix, and it can even do threads.

**Reply by cipherboy** [reply]
As another alternative to Slack, I've heard good things about [Zulip](https://zulip.com/) (its [open source](https://github.com/zulip/zulip/)) -- it supports GitHub auth (along with others) so users wouldn't feel as tied to creating a new account... And has [sponsorship for OSS](https://zulip.com/plans/). Threading and LaTeX support too. [Looks like there's Matrix support](https://zulip.com/integrations/doc/matrix) too.

**Reply by matejak** [reply]
It seems that Gitter is favoured, as it is also FOSS and speaks Matrix, but it has an IRC bridge as well: https://irc.gitter.im/
